app/controllers/user_notifications_controller.rb

```diff
@@ -21,4 +21,14 @@ def delete
     ).destroy!
     render nothing: true
   end
+
+  def unsubscribe
+    user = User.find_by(email: params[:token][16,params[:token].length])
```

Comment:
Is the intention of this is to make the token in the format of `#{token}#{email}`?

May I suggest just having a token param instead and do something like `UserNotification.find_by(token: params[:token])`? You would need to make the token unique per row in this case.

Reply:
Done